homeassistant/components/ruckus_unleashed/config_flow.py

```diff
@@ -26,9 +34,18 @@ async def validate_input(hass: core.HomeAssistant, data):
     except ConnectionError as error:
         raise CannotConnect from error
 
-    mesh_name = ruckus.mesh_name()
+    mesh_name = await hass.async_add_executor_job(ruckus.mesh_name)
```

Comment:
Since we have multiple calls that needs the executor, I would make the whole `validate_input` function sync and schedule it on the executor. That way we save some context switches, which are expensive.

Reply:
That makes sense. Fixed.

There's quite a few functions scheduled on the executor in `async_setup_entry`, too. Would you prefer that I pull those out into a separate function or should I leave them in `async_setup_entry`? It seems more standard to leave them in `async_setup_entry`.

Reply:
It can be a future refactor. Maybe by creating a class that stores the different info and provides a method to fetch it.

Reply:
Sounds good. I plan to add async support to my external `pyruckus` library in a future update, too. Thanks!